shiny/ui/_chat_normalize.py

```diff
@@ -26,11 +25,11 @@ class GenerateContentResponse:
 
 class BaseMessageNormalizer(ABC):
     @abstractmethod
-    def normalize(self, message: Any) -> ChatMessage:
+    def normalize(self, message: Any) -> ChatUIMessage:
```

Comment:
It's hard to see the boundary of where a `ChatMessage` crosses into being a `ChatUIMessage` but most uses of `ChatUIMessage` seems like it should be a `ChatMessage`. 

What benefit do we get by maintaining two types of chat message? Once a raw message is handed to the chat component, isn't it by definition a `ChatUIMessage`?

Reply:
> holding all the logic and internal data that it needs without leaking that into the public API.

What are we trying to keep out of the public API? There's a tiny amount of work in the `ChatUIMessage` constructor that prepares the chat message for use in Shiny, but that seems entirely appropriate for a `ChatMessage`/`ChatUIMessage` class.

I don't see anything in this PR that strikes me as reason we'd want to have an internal-facing class that we hide from the public API, but maybe you have other ideas in mind?

Reply:
See ([3bc17be](https://github.com/posit-dev/py-shiny/pull/1897/commits/3bc17beea2991fdc12dddf09c1c8d1168d7f5b20)